internal/ingest/linksystem.go

```diff
@@ -258,6 +253,7 @@ func (ing *Ingester) storageHook(pubID peer.ID, c cid.Cid) {
 			lnk, err := ad.FieldPreviousID().Must().AsLink()
 			if err != nil {
 				log.Errorw("Cannot read previous link from advertisement", "err", err)
+				return
```

Comment:
The return was intentionally absent so that if unable to read the previous link, we would still process this ad.  If you agree, then remove return.  If you want to skip processing this ad, then remove the following `else`.

Reply:
I want to skip processing this ad. I think it's okay to not process an ad that is malformed and if this has a field for previous ID *and* it fails to parse as a link then it is malformed.